### PR TITLE
Stateful-pallet: added rpc support to get pages

### PR DESCRIPTION
**Pull request by aramikm**
# Goal
The goal of this PR is to implement a unified RPC endpoint to get Itemized and paginated stored items.

Closes issue #815 

# Discussion
- Same RPC exposes all Itemized and Paginated Items (we might want to filter them in future but no required usecase currently)
- Modifies Child tree to use TwoX hash instead of identity
- Clean ups

# Checklist
- [X] Custom RPC OR Runtime API added/changed? Updated js/api-augment. 
- [x] Tests added

### CODE REVIEW

File: pallets/stateful-storage/src/lib.rs [hunk 1]
```diff
@@ -274,20 +277,84 @@ pub mod pallet {
...
+							},
+						}
+					}
+				},
+				PayloadLocation::Paginated => {
+					let keys: Vec<StatefulPageKeyPart> = vec![schema_id.encode().to_vec()];
+					let mut pages: Vec<PageResponse> = StatefulChildTree::prefix_iterator::<
+						Page<T::MaxPaginatedPageSizeBytes>,
+					>(&msa_id, &keys[..])
+					.filter_map(|(key, v)| match <u16>::decode(&mut &key[..]) {
```
Comment (aramikm) [inline pallets/stateful-storage/src/lib.rs:326]: u16 should be replaced with `PageId` type after merge

File: pallets/stateful-storage/src/lib.rs [hunk 2]
```diff
@@ -274,20 +277,84 @@ pub mod pallet {
...
-			page_response
+impl<T: Config> Pallet<T> {
+	/// This function returns all the pages associated with `msa_id` and `schema_id` params
+	/// and support both types of storage types
+	/// Itemized - For this storage it parses all the items and returns each item with its index
+	/// Paginated - For this storage it finds all the pages and return each page with its pageId
+	///
+	/// [Warning] since this function iterates over all the potential keys it should never called
+	/// from runtime.
+	pub fn get_pages(msa_id: MessageSourceId, schema_id: SchemaId) -> Vec<PageResponse> {
```
Comment (aramikm) [inline pallets/stateful-storage/src/lib.rs:290]: This does not return any error and just ignores any if happened and return only valid items.
Let me know if you think it makes sense to expose errors also.

File: pallets/stateful-storage/src/lib.rs [hunk 3]
```diff
@@ -234,10 +240,7 @@ pub mod pallet {
 
 			match updated_page.is_empty() {
 				true => {
-					StatefulChildTree::kill::<Page<T::MaxItemizedPageSizeBytes>>(
-						&state_owner_msa_id,
-						&keys,
-					);
+					StatefulChildTree::kill(&state_owner_msa_id, &keys);
 					Self::deposit_event(Event::PageRemoved {
```
Comment (saraswatpuneet) [inline pallets/stateful-storage/src/lib.rs:244]: :+1:

File: pallets/stateful-storage/src/lib.rs (hunk 2 above)
Comment (saraswatpuneet) [reply pallets/stateful-storage/src/lib.rs:290]: Imo: its ok like this, if request is legit! then they would expect results else empty , we follow this pattern at quite a few places, but not opposed to errors if they will actually be meaningful

**Review by saraswatpuneet — APPROVED** [review]
:+1:  can we put a TODO comment for changes needed to be done for post merge like

 <u16> to <PageId>

File: pallets/stateful-storage/src/lib.rs (hunk 2 above)
Comment (JoeCap08055) [reply pallets/stateful-storage/src/lib.rs:290]: I agree. No need to return an error unless there's something meaningful or critical.

**Review by saraswatpuneet — APPROVED** [review]
👍

**Pull request merged**